src/main/java/uk/gov/justice/digital/hmpps/keyworker/rolemigration/RoleAssignmentsService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.justice.digital.hmpps.keyworker.rolemigration;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.client.HttpClientErrorException;
+import uk.gov.justice.digital.hmpps.keyworker.dto.RoleAssignmentsSpecification;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Validated
+public class RoleAssignmentsService {
+
+    private final RoleService roleService;
+
+    public RoleAssignmentsService(RoleService roleService) {
+        this.roleService = roleService;
+    }
+
+    /**
+     * Look up users by caseload (prison) and membership of specification.rolesToMatch.
+     * for each matched user:
+     * Assign to the user the roles identified in specification.rolesToAssign.
+     * Remove from the user (at the caseload) the roles identified by specification.rolesToRemove.
+     */
+    @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES', 'MAINTAIN_ACESS_ROLES_ADMIN')")
```

Comment:
`MAINTAIN_ACESS_ROLES_ADMIN` should be `ACCESS`?

Reply:
Don't think MAINTAIN_ACCESS_ROLES should be allowed